instagan/train.py

```diff
@@ -58,17 +81,52 @@
                 save_suffix = "iter_%d" % total_steps if opt.save_by_iter else "latest"
                 model.save_networks(save_suffix)
 
-            iter_data_time = time.time()
+            iter_data_time = time.time()      
         if epoch % opt.save_epoch_freq == 0:
             print(
                 "saving the model at the end of epoch %d, iters %d"
                 % (epoch, total_steps)
             )
             model.save_networks("latest")
             model.save_networks(epoch)
-
+        
         print(
             "End of epoch %d / %d \t Time Taken: %d sec"
             % (epoch, opt.niter + opt.niter_decay, time.time() - epoch_start_time)
         )
-        model.update_learning_rate()
+        
+        # INFERENCE CODE 
+        try:os.mkdir(opt_test.results_dir+'epoch'+str(epoch)+'/')
+        except:pass
```

Comment:
Exceptions should be handled. I expect you expect only sir already exists but other stuff may come up, at least print

Reply:
Exception handled with new commt